core/include/prometheus/registry.h

```diff
@@ -36,6 +36,20 @@ namespace prometheus {
 /// a data race.
 class Registry : public Collectable {
  public:
+  /// \brief How to deal with repeatedly added family names for a type
+  enum class InsertBehavior {
+    /// \brief Create new family object and append
```

Comment:
Can you describe more verbosely how the algorithm works? What happens when there is no existing family with that name, what happens when there is an existing family with that name, but a different type, etc?

Reply:
ie in the doc comment for each behavior